src/soap.ts

```diff
@@ -233,6 +234,21 @@ export class SOAP<S extends Schema> extends HttpApi<S> {
   /** @override */
   beforeSend(request: HttpRequest & { _message: object }) {
```

Comment:
I'm not a fan of this design (pass an object for mutation).  I know you didn't start it.

But if http-api isn't exported and is only used via the 2 classes that extend it (soap and bulk?) maybe it's possible to fix it?

Reply:
Or maybe that's too big and it's easier to stay in line with what jsforce is already doing?